packages/@glimmer/core/src/owner.ts

```diff
@@ -0,0 +1,27 @@
+import { DEBUG } from '@glimmer/env';
+
+const OWNER_MAP = new WeakMap<{}, unknown>();
+
+export const OWNER_KEY = `__OWNER_${Math.floor(Math.random() * Date.now())}__`;
+
+export let DEFAULT_OWNER = {};
+
+if (DEBUG) {
+  const OWNER_ERROR = 'You attempted to use the Owner for a component, modifier, or helper, but did not provide an owner to `renderComponent`.';
+  DEFAULT_OWNER = new Proxy(DEFAULT_OWNER, {
+    get(): never {
+      throw new Error(OWNER_ERROR);
+    },
+    set(): never {
+      throw new Error(OWNER_ERROR);
+    }
+  });
+}
+
+export function getOwner<T = unknown>(obj: object): T {
```

Comment:
I don't think it will matter too much, but `obj` just needs to be a valid weakmap key and that _would_ allow functions.

Reply:
I believe `object` works this way as a type: https://www.typescriptlang.org/play/index.html?ssl=3&ssc=17&pln=1&pc=1#code/DYUwLgBA9gRgVgLmvEBjMBuAsAKF7OCAXggAoBKYgPggG8IBfDIA

Reply:
WAT, 😞 sorry I didn't realize that